databricks/koalas/frame.py

```diff
@@ -532,10 +530,20 @@ def count(self):
         Age       4
         Single    5
         dtype: int64
-
-        Counts for each **row**:
         """
-        return self._sdf.count()
+        # Build the expressions to do counting.
+        count_exprs = []
+        for col in self._metadata.column_fields:
+            spark_type = self._sdf.schema[col].dataType
+            # Special handle floating point types because Spark's count treats nan as a valid value,
+            # whereas Pandas count doesn't include nan.
+            if isinstance(spark_type, DoubleType) or isinstance(spark_type, FloatType):
+                count_exprs.append(F.count(F.nanvl(col, F.lit(None))).alias(col))
+            else:
+                count_exprs.append(F.count(col).alias(col))
+
+        row = self._sdf.select(*count_exprs).collect()[0]
+        return pd.Series(row.asDict())
```

Comment:
We might be able to reuse `DataFrame._reduce_for_stat_function`.
The `NaN` handling is slightly different but I guess we should do the same thing in the `_reduce_for_stat_function`, and then we can reuse it.

Reply:
I agree, `_reduce_for_stat_function` should be the standard path.

Reply:
OK in the future please add docstring comment for every function. It took me a while to understand what _reduce_for_stat_function does.

Reply:
_reduce_for_stat_function passes a spark.Column in, and there is no way to get the type. Should I refactor that to pass also the column data type in?